src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
I mean that it is probably better to have the controller-facing API to contain async (Future-returning) methods only so to avoid a possible case when such a method is also returned as a "regular" (not Async) action to prevent blocking of thread pool.

That might be my misunderstanding, but there are two types of request handling in Play - Sync and Async Ones?
If Play always wraps every request into an Async action, that's not an issue and just me being too pessimistic.